server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -710,10 +717,17 @@ public void insert(InsertRowPlan insertRowPlan) throws WriteProcessException {
    */
   @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
   public void insertTablet(InsertTabletPlan insertTabletPlan) throws BatchInsertionException {
+    TSStatus[] results = new TSStatus[insertTabletPlan.getRowCount()];
```

Comment:
I think it's better to move the two line after blockInsertionIfReject  for that we need not to create them before we get memory.

Reply:
Sure~